### PR TITLE
Update the HMQV key derivation details. Clean up some presentation bugs.

### PR DESCRIPTION
**Pull request by chris-wood**

### CODE REVIEW

File: draft-irtf-cfrg-opaque.md [hunk 1]
```diff
@@ -1145,33 +1149,75 @@ gets the most significant bytes of the HKDF output, Km2 the next bytes, etc.
...
+~~~
 
-   and u = H(epkU \| "client" \| info) and s = H(epkS \| "server" \| info).
+where `s` is computed as follows:
 
-<!--     (u and s are often denoted d and e)   -->
+~~~
+hashInput = len(epkS) || epkS ||
+            len(info) || info ||
+            len("server") || "client"
```
Comment (hugokraw) [inline draft-irtf-cfrg-opaque.md:1194]: change to:
len("server") || "server"

File: draft-irtf-cfrg-opaque.md [hunk 2]
```diff
@@ -1145,33 +1149,75 @@ gets the most significant bytes of the HKDF output, Km2 the next bytes, etc.
...
+s = Hash(hashInput) mod p
```
Comment (hugokraw) [inline draft-irtf-cfrg-opaque.md:1195]: change mod p to mod (len(p)-1)

File: draft-irtf-cfrg-opaque.md [hunk 3]
```diff
@@ -1145,33 +1149,75 @@ gets the most significant bytes of the HKDF output, Km2 the next bytes, etc.
...
+~~~
 
 #### 3DH key derivation
 
- - info = "3DH keys" \| nonceU \| nonceS \| IdU \| IdS
+The HKDF input parameter `info` is computed as follows:
```
Comment (hugokraw) [inline draft-irtf-cfrg-opaque.md:1200]: remove last line

File: draft-irtf-cfrg-opaque.md [hunk 4]
```diff
@@ -1145,33 +1149,75 @@ gets the most significant bytes of the HKDF output, Km2 the next bytes, etc.
 
 Values IKM and info are defined for each protocol:
 
-#### HMQV key derivation
+#### HMQV key derivation {#derive-3dh}
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md:1152]: Oops, wrong headers!

File: draft-irtf-cfrg-opaque.md (hunk 1 above)
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md:1194]: hah, I thought I copied the right text!

File: draft-irtf-cfrg-opaque.md (hunk 3 above)
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md:1200]: Which line?

Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md:1200]: (I wasn't sure what line you were referring to, so I'll merge now and can clean this up later!)

Comment (hugokraw) [reply draft-irtf-cfrg-opaque.md:1200]: My bad. I read HMQV instead of HKDF (too many H's :-)

**Pull request merged**